docs/how-to-guides/editable-installs.rst

```diff
@@ -0,0 +1,79 @@
+.. _how-to-guides-editable-installs:
+
+*****************
+Editable installs
+*****************
+
+We have support for editable installs! To use it, simply pass the
+``-e``/``--editable`` option to pip_.
+
+
+.. code-block:: console
+
+   $ pip install -e .
+
+
+This will install the project in the current directory in editable mode. When a
+``meson-python`` package is installed in editable mode, it will be re-built the
+first time it is imported. That means, the first time you import a module from
+your package, it will take a bit longer.
+
+
+.. admonition:: What is the scope of editable installs?
+   :class: seealso
+
+   The editable installs feature only targets the Python modules of the project.
```

Comment:
This sounds like pure Python. It's worth pointing out that, unlike in any other build backend, files for an extension module (C/C++/Cython/whatever) are also editable and will be automatically rebuilt.